06-aks-cluster.md

```diff
@@ -9,8 +9,11 @@ Now that your [ACR instance is deployed and ready to support cluster bootstrappi
    > If you cloned this repo, then the value will be the original mspnp GitHub organization's repo, which will mean that your cluster will be bootstraped using public container images. If instead you forked this repo, then the GitOps repo will be your own repo, and your cluster will be bootstrapped using container images references based on the values in your repo's manifest files. On the prior instruction page you had the oppertunity to update those manifests to use your ACR instance. For guidance on using a private bootstrapping repo, see [Private bootstrapping repository](./cluster-manifests/README.md#private-bootstrapping-repository).
 
    ```bash
-   GITOPS_REPOURL=$(git config --get remote.origin.url)
-   echo GITOPS_REPOURL: $GITOPS_REPOURL
+   GITOPS_REPOURL_AKS_BASELINE=$(git config --get remote.origin.url)
+   echo GITOPS_REPOURL_AKS_BASELINE: $GITOPS_REPOURL_AKS_BASELINE
+
+   GITOPS_CURRENT_BRANCH_NAME_AKS_BASELINE=$(git branch --show-current)
+   echo GITOPS_CURRENT_BRANCH_NAME_AKS_BASELINE: $GITOPS_CURRENT_BRANCH_NAME_AKS_BASELINE
```

Comment:
Why end them in `_AKS_BASLINE` if they are not going to be `export`ed?

Reply:
That's true. I have lost the `export` when copying from another "one shot script" where I had it. Will add `export`.

Reply:
are you sure it's needed?  I didn't catch where you ended up needing it later.

Reply:
You are right, it will only be needed in the following step if you strictly follow the process.

But I did a couple of deployment (especially while I was adding the name prefixes and chasing this role lookup bug) and did that step over and over again. Since I had multiple terminals open, I switched context by running `saveenv.sh` (which is lovely -- never seen that before!) and sourcing the `aks_baseline.env` to restore the context. 

Later, another deployment failed because the branch variable that I had added was not passed correctly and so, the bootstrapping tried to get scripts from the main branch that had not the replaced repo FQDNs.

This is why I thought (just for consistency) it might make sense to also put it into the environment.

Reply:
Gotcha.  Yeah, our policy so far has been that we only export the variable if it's going to be used on a "later page" in the steps.  The pages are pretty natural 'stopping points' for someone to go wander away so higher likelihood of a console blip (Looking at you, codespaces!).  So that was the rule we put into place, never on ones where the command is used just on the same page (usually as the very next step).

Reply:
Thanks for the feedback on the `saveenv.sh` -- that was born out of codespaces dropping the console frequently.  The Commercial Software Engineering crew came up with this for the codespaces-based workshops they run with the AKS baseline.  It's very handy.

Reply:
So you prefer to have export and suffix removed?
(As far as I see, these are the only two variables without export and suffix...)

Reply:
Oh now -- there are some more.

Reply:
Hmmm... Well, I love the idea of not needing to think about whether a variable will be written to the env file or not. But this is a personal preference and I'll be fine with whatever you decide.

Reply:
Correct,  the only variables that should be `export`'d are ones that will be accessed on a future page.

Examples of ones that are not exported can be found on
- the AAD page (only used on that page)
- The networking page (only used on that page)
- This GITOPS url (and now branch) 
- The Bootstrap validation page 
- etc

Bascially only those that have `_AKS_BASELINE` should have `export` and the only variables that should have `_AKS_BASELINE` are those that are used on a future page.

Reply:
This is consistent with the walkthroughs on the AKS Baseline for multi-region and AKS Baseline for PCI-DSS 3.2.1 (regulated).  Any chance in this policy should probably be a backlog item and addressed across them all.  But for now this pattern has been met with success.

Reply:
Understood. 👍

Reply:
@ulkeba -- in your revert of this, you missed a spot that you had changed lower on the page.

Reply:
Good catch.
Sorry.

Reply:
Resolved!